morf-core/src/main/java/org/alfasoftware/morf/upgrade/GraphBasedUpgradeScriptGenerator.java

```diff
@@ -73,17 +81,18 @@ public List<String> generatePreUpgradeStatements() {
     // temp table
     statements.addAll(connectionResources.sqlDialect().tableDeploymentStatements(idTable));
 
+    statements.addAll(UpgradeHelper.preSchemaUpgrade(sourceSchema,
+            targetSchema,
+            viewChanges,
+            viewChangesDeploymentHelperFactory.create(connectionResources)));
+
     // drop views
     for (View view : viewChanges.getViewsToDrop()) {
-      // non-present views can be listed amongst ViewsToDrop due to how we calculate dependencies
-      if (sourceSchema.viewExists(view.getName())) {
-        statements.addAll(connectionResources.sqlDialect().dropStatements(view));
-      }
       if (sourceSchema.tableExists(DEPLOYED_VIEWS_NAME) && targetSchema.tableExists(DEPLOYED_VIEWS_NAME)) {
         statements.addAll(ImmutableList.of(
                 connectionResources.sqlDialect().convertStatementToSQL(
-            delete(tableRef(DEPLOYED_VIEWS_NAME)).where(Criterion.eq(field("name"), view.getName().toUpperCase()))
-          )));
+                        delete(tableRef(DEPLOYED_VIEWS_NAME)).where(Criterion.eq(field("name"), view.getName().toUpperCase()))
+                )));

```

Comment:
This should be using `ViewChangesDeploymentHelper.dropViewIfExists()`,
insted of directy `delete(tableRef(DEPLOYED_VIEWS_NAME))`